docs/best-practices/code-reviews/readme.md

```diff
@@ -146,214 +183,256 @@ Further paragraphs come after blank lines.
   current code?
 
+- **Do you see potential for useful abstractions?** Partially duplicated code
+  often indicates that a more abstract or general piece of functionality
+  can be extracted and then reused in different contexts.
+
 - **Think like an adversary.** Try to "catch" authors taking shortcuts or
   missing cases by coming up with problematic configurations/input
   data that breaks their code.
 
 - **Think about libraries or existing product code.** When someone
   re-implements existing functionality, more often than not it's
-  simply because they don't know it already exists. If it's
-  purposeful, adding a comment to justify it is usually helpful. As a
-  bonus, vetted libraries often have opinions on structure/design that
-  are backed by experience.
+  simply because they don't know it already exists. Sometimes, code
+  or functionality is duplicated on purpose, e.g., in order to avoid
+  dependencies. In such cases, a code comment can clarify the intent.
+  Is the introduced functionality already provided by an existing library?
+
+- **Does the change follow standard patterns?** Established code bases
+  often exhibit patterns around naming conventions, program logic
+  decomposition, data type definitions, etc. It is usually desirable
+  that changes are implemented in accordance with existing patterns.
 
-- **Does this add compile-time or run-time dependencies (especially
+- **Does the change add compile-time or run-time dependencies (especially
   between sub-projects)?** We want to keep our products loosely coupled,
-  with as few dependencies as possible. Any build.gradle changes
-  should be scrutinized heavily.
+  with as few dependencies as possible. Changes to dependencies and the
+  build system should be scrutinized heavily.
 
-### Legibility and Style
+### Legibility and style
 
 - **Think about your reading experience.** Did you grasp the concepts in a
-  reasonable amount of time? Was the flow sane and names easy to
-  follow? Were you able to keep track through multiple files or
-  functions? Were you put off by inconsistent naming?
+  reasonable amount of time? Was the flow sane and were variable and methods names
+  easy to follow? Were you able to keep track through multiple files or functions?
+  Were you put off by inconsistent naming?
 
 - **Does the code adhere to coding guidelines and code style?** Is the
   code consistent with the project in terms of style, API conventions,
-  etc?
+  etc.? As mentioned above, we prefer to settle style debates with automated
+  tooling.
 
 - **Does this code have TODOs?** TODOs just pile up in code, and become
-  stale over time. Have the author submit a ticket on Github Issues and
-  attach the issue number to the TODO. Similarly, the CR should not have
-  large amounts of commented-out code.
+  stale over time. Have the author submit a ticket on GitHub Issues or JIRA and
+  attach the issue number to the TODO. The CR should not contain
```

Comment:
nit: it's not the _CR_ with commented code, it's the "proposed code change" or "patch"

Reply:
done